moto/iot/models.py

```diff
@@ -883,7 +883,11 @@ def update_thing(
             if not do_merge:
                 thing.attributes = attributes
             else:
-                thing.attributes = {k: v for k, v in attributes.items() if v}
+                for k, v in attributes.items():
+                    if v:
+                        thing.attributes[k] = v
+                    else:
+                        thing.attributes.pop(k)
```

Comment:
How does AWS behave if `k` does not exist here?
I.e., thing.attributes only has attribute`k1`, and a user calls `update_thing(.. {"k2": ""})`?

Reply:
No exception, nothing changed. Simply a `{'ResponseMetadata': {..., 'HTTPStatusCode': 200,...}`.

Reply:
I just check another special case.

With `"merge": False` and `update_thing(.. {"k2": ""})`, all attributes are deleted. So an empty value will never be saved.